dev/io.openliberty.microprofile.reactive.messaging.internal/src/io/openliberty/microprofile/reactive/messaging/internal/RMAsyncProviderFactoryImpl.java

```diff
@@ -0,0 +1,183 @@
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.osgi.service.component.annotations.ReferencePolicyOption;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.wsspi.threadcontext.ThreadContext;
+import com.ibm.wsspi.threadcontext.ThreadContextDescriptor;
+import com.ibm.wsspi.threadcontext.WSContextService;
+
+import io.openliberty.microprofile.reactive.messaging.internal.interfaces.RMAsyncProvider;
+import io.openliberty.microprofile.reactive.messaging.internal.interfaces.RMAsyncProviderFactory;
+import io.openliberty.microprofile.reactive.messaging.internal.interfaces.RMContext;
+
+@Component
+public class RMAsyncProviderFactoryImpl implements RMAsyncProviderFactory {
+
+    private static final TraceComponent tc = Tr.register(RMAsyncProviderFactoryImpl.class);
+
+    @SuppressWarnings("unchecked")
+    private static final Map<String, ?>[] DEFAULT_CONTEXT_PROVIDERS = new Map[] {
+                                                                                  Collections.singletonMap(WSContextService.THREAD_CONTEXT_PROVIDER,
+                                                                                                           "com.ibm.ws.classloader.context.provider"),
+                                                                                  Collections.singletonMap(WSContextService.THREAD_CONTEXT_PROVIDER,
+                                                                                                           "com.ibm.ws.javaee.metadata.context.provider"),
+                                                                                  Collections.singletonMap(WSContextService.THREAD_CONTEXT_PROVIDER,
+                                                                                                           "com.ibm.ws.security.context.provider"),
+    };
+
+    /**
+     * The built-in context service
+     */
+    @Reference(target = "(service.pid=com.ibm.ws.context.manager)", policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+    private volatile WSContextService builtInContextService;
+
+    /**
+     * The default context service provided by the concurrent feature
+     */
+    @Reference(target = "(id=DefaultContextService)", cardinality = ReferenceCardinality.OPTIONAL, policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+    private volatile WSContextService defaultContextService;
+
+    private final Map<String, WSContextService> namedContextServices = Collections.synchronizedMap(new HashMap<>());
+
+    /**
+     * The liberty global thread pool executor
+     */
+    @Reference(target = "(component.name=com.ibm.ws.threading)", policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
+//    @Reference(service = ManagedExecutorService.class, policy = ReferencePolicy.DYNAMIC, policyOption = ReferencePolicyOption.GREEDY)
```

Comment:
Is this commented out item needed, looks like it was has been replaced?

Reply:
This is left over from when I was tracking down the cause of a behaviour change when using the global thread pool vs. `ManagedExecutorService`. I will remove it.